src/filter.rs

```diff
@@ -269,54 +269,87 @@ mod simd {
 /// this does not operate on pixels but on raw bytes of a scanline.
 ///
 /// Details on how each filter works can be found in the [PNG Book](http://www.libpng.org/pub/png/book/chapter09.html).
+///
+/// The default filter is `Adaptive`, which uses heuristics to select the best filter for every row.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum Filter {
+    NoFilter,
+    Sub,
+    Up,
+    Avg,
+    Paeth,
+    Adaptive,
+}
+
+impl Default for Filter {
+    fn default() -> Self {
+        Filter::Adaptive
+    }
+}
+
+impl From<RowFilter> for Filter {
+    fn from(value: RowFilter) -> Self {
+        match value {
+            RowFilter::NoFilter => Filter::NoFilter,
+            RowFilter::Sub => Filter::Sub,
+            RowFilter::Up => Filter::Up,
+            RowFilter::Avg => Filter::Avg,
+            RowFilter::Paeth => Filter::Paeth,
+        }
+    }
+}
+
+impl Filter {
+    pub(crate) fn from_simple(compression: Compression) -> Self {
+        match compression {
+            Compression::NoCompression => Filter::NoFilter, // with no DEFLATE filtering would only waste time
+            Compression::Fastest => Filter::Up, // fast and avoids long backreferences in DEFLATE stream
```

Comment:
Is this always `Up`? No filter may compress better, so this could be adaptive that selects between None and Up.

Which makes me think that `Filter::Adaptive` is not sufficiently configurable. Filtering can have many strategies and try different sets of filters.

Reply:
Yeah, `Up` is probably not a good choice for `Fastest` mode. It's not actually the fastest. I'll switch it over to `None`, good call.

Reply:
https://github.com/image-rs/image-png/pull/503#discussion_r1872203847